awx/main/access.py

```diff
@@ -853,7 +853,12 @@ class InventoryAccess(BaseAccess):
     """
 
     model = Inventory
-    prefetch_related = ('created_by', 'modified_by', 'organization')
+    prefetch_related = (
+        'created_by',
+        'modified_by',
+        'organization',
+        Prefetch('labels', queryset=Label.objects.all().order_by('name')),
+    )
```

Comment:
In the access class `can_attach` and `can_unattach` should get called when managing labels... but it's likely the case that `BaseAccess` logic is sufficient for this.